internal/dag/conditions.go

```diff
@@ -179,3 +179,10 @@ func headerMatchConditionsValid(conditions []contour_api_v1.MatchCondition) erro
 
 	return nil
 }
+
+// ValidateRegex returns an error if the supplied
+// regex is invalid.
+func ValidateRegex(regex string) error {
```

Comment:
Nit: Best to add a comment that this is specifically validating RE2 regex syntax because that I what we are programming into Envoy (though Go isn't *identical*l apparently?).